pyanaconda/modules/network/firewall/firewall.py

```diff
@@ -0,0 +1,153 @@
+        self.set_disabled_services(data.firewall.remove_services)
+
+    def setup_kickstart(self, data):
+        """Setup the kickstart data."""
+        data.firewall.use_system_defaults = self.use_system_defaults
+        data.firewall.enabled = self.firewall_enabled
+        data.firewall.ports = self.enabled_ports
+        data.firewall.trusts = self.trusts
+        data.firewall.services = self.enabled_services
+        data.firewall.remove_services = self.disabled_services
+        return data
+
+    @property
+    def firewall_seen(self):
+        return self._firewall_seen
+
+    def set_firewall_seen(self, firewall_seen):
+        self._firewall_seen = firewall_seen
+        self.firewall_seen_changed.emit()
+        log.debug("Firewall command considered seen in kickstart: %s.", self._firewall_seen)
+
+    @property
+    def use_system_defaults(self):
+        return self._use_system_defaults
+
+    def set_use_system_defaults(self, use_system_defaults):
+        self._use_system_defaults = use_system_defaults
+        self.use_system_defaults_changed.emit()
+        if self.use_system_defaults:
+           log.debug("System default will be used for firewall.")
+        else:
+           log.debug("Anaconda will configure firewall.")
+
+    @property
+    def firewall_enabled(self):
+        return self._firewall_enabled
+
+    def set_firewall_enabled(self, firewall_enabled):
+        self._firewall_enabled = firewall_enabled
+        self.firewall_enabled_changed.emit()
+        if self.firewall_enabled:
+            log.debug("Firewall will be enabled.")
+        else:
+            log.debug("Firewall will be disabled.")
+
+    @property
+    def enabled_ports(self):
+        return self._enabled_ports
+
+    def set_enabled_ports(self, enabled_ports):
+        self._enabled_ports = list(enabled_ports)
+        self.enabled_ports_changed.emit()
+        log.debug("Ports that will be allowed through the firewall: %s", self._enabled_ports)
+
+    @property
+    def trusts(self):
+        return self._trusts
+
+    def set_trusts(self, trusts):
+        self._trusts = list(trusts)
+        self.trusts_changed.emit()
+        log.debug("Trusted devices that will be allowed through the firewall: %s", self._trusts)
+
+    @property
+    def enabled_services(self):
+        return self._enabled_services
+
+    def set_enabled_services(self, enabled_services):
+        self._enabled_services = list(enabled_services)
+        self.enabled_services_changed.emit()
+        log.debug("Services that will be allowed through the firewall: %s", self._enabled_services)
+
+    @property
+    def disabled_services(self):
+        return self._disabled_services
+
+    def set_disabled_services(self, disabled_services):
+        self._disabled_services = list(disabled_services)
+        self.disabled_services_changed.emit()
+        log.debug("Services that will be explicitly disabled on the firewall: %s", self._disabled_services)
```

Comment:
Please wrap this line. I guess it is more than 99 characters.

Reply:
Wrapped.